cpp/include/raft/stats/minmax.cuh

```diff
@@ -68,6 +70,73 @@ void minmax(const T* data,
     data, rowids, colids, nrows, ncols, row_stride, globalmin, globalmax, sampledcols, stream);
 }
 
+/**
+ * @brief Computes min/max across every column of the input matrix, as well as
+ * optionally allow to subsample based on the given row/col ID mapping vectors
+ *
+ * @tparam T Data type of input matrix element.
+ * @tparam IndexType Index type of matrix extent.
+ * @tparam LayoutPolicy Layout type of the input matrix. When layout is strided, it can
+ *                      be a submatrix of a larger matrix. Arbitrary stride is not supported.
+ * @tparam AccessorPolicy Accessor for the input and output, must be valid accessor on
+ *                        device.
+ * @tparam TPB number of threads per block
+ * @param handle the raft handle
+ * @param data input data col-major of size [nrows, ncols], unless rowids or
+ * colids length is smaller
+ * @param rowids actual row ID mappings. It is of length nrows. If you want to
+ * skip this index lookup entirely, pass nullptr
+ * @param colids actual col ID mappings. It is of length ncols. If you want to
+ * skip this index lookup entirely, pass nullptr
+ * @param globalmin final col-wise global minimum (size = ncols)
+ * @param globalmax final col-wise global maximum (size = ncols)
+ * @param sampledcols output sampled data. Pass nullptr if you don't need this
+ * @note This method makes the following assumptions:
+ * 1. input and output matrices are assumed to be col-major
+ * 2. ncols is small enough to fit the whole of min/max values across all cols
+ *    in shared memory
+ */
+template <typename T,
+          typename IdxType,
+          typename LayoutPolicy,
+          typename AccessorPolicy,
+          int TPB = 512>
```

Comment:
Would you consider letting users pass in the value of `TPB` as a `std::integral_constant<int, TPB>` parameter?  That way, users wouldn't have to call this function as `minmax<TPB>(...)`, and they could pass down the integral constant through a chain of nested function calls.

```
template <typename T,
          typename IdxType,
          typename LayoutPolicy,
          typename AccessorPolicy,
          int TPB = 512>
Styling with Markdown is supported
void minmax(const raft::handle_t& handle,
  /* the other parameters */,
  std::integral_constant<int, TPB> /* threads_per_block */);
```